code/modules/projectiles/gun.dm

```diff
@@ -142,6 +142,8 @@
 
 	if(pin && needspin)
 		pin = new pin(src)
+	else if(pin)
+		pin = null
```

Comment:
Can this just be new'd on line 141? It's kind of weird for the `pin` var to contain a path and a ref to a pin obj instance.

We should always have a pin on line 143 if `needspin == TRUE`, due to the check on line `140`.